operational_analysis/methods/turbine_long_term_gross_energy.py

```diff
@@ -126,6 +124,30 @@ def run(self, enable_plotting = False,
         self.enable_plotting = enable_plotting 
         self.plot_dir = plot_dir
         
+        self._reanal = reanal_subset # Reanalysis data to consider in fitting
+        
+        # Check uncertainty types
+        if self.UQ == False:
+            if type(wind_bin_thresh) != float:
+                logger.info("wind_bin_thresh needs to be a float if UQ is False!")
+            if type(max_power_filter) != float:
+                logger.info("max_power_filter needs to be a float if UQ is False!")
+            if type(correction_threshold) != float:
+                logger.info("correction_threshold needs to be a float if UQ is False!")
+        if self.UQ == True:
+            if type(wind_bin_thresh) != tuple:
+                logger.info("wind_bin_thresh needs to be a tuple if UQ is True!")
+            if type(max_power_filter) != tuple:
+                logger.info("max_power_filter needs to be a tuple if UQ is True!")
+            if type(correction_threshold) != tuple:
+                logger.info("correction_threshold needs to be a tuple if UQ is True!")        
+        # Define relevant uncertainties, to be applied in Monte Carlo sampling
```

Comment:
Shouldn't an error be thrown in this case? You can use assert here instead of logger.info:
```
if type(wind_bin_thresh) != float:
    logger.info("wind_bin_thresh needs to be a float if UQ is False!")
```
to
```
assert type(wind_bin_thresh) == float, "wind_bin_thresh needs to be a float if UQ is False!"
```

I don't have a problem with this, even though it's a little repetitive. Input checking code is difficult to shorten without losing clarity. For your own information, If you want to make this shorter, you coule do something like this:
```
vars = [wind_bin_thresh, max_power_filter, correction_threshold]
expected_type = float if self.UQ == True else tuple
if: any([type(x) != expected_type for x in vars]):
    logger.info("...")
```

Reply:
Cool thanks!